### ISSUE TITLE
[Bug]: Child Process Forked from Module Does Not Support process.send in Electron v28.1.2+

### ISSUE THREAD

**Issue by seydx** [electron/electron#40987]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

28.1.3

### What operating system are you using?

macOS

### Operating System Version

Sonoma 14.2.1

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

28.1.1

### Expected Behavior

In previous versions of Electron, Module B was able to communicate back to Module A using process.send(). The expected behavior is that after forking Module B from Module A, Module B should be able to use process.send() to communicate with Module A.

Steps to Reproduce
- Fork a module (Module B) from another module (Module A) using child_process.fork().
- In Module B, attempt to send a message back to Module A using process.send('started').
- process.send is undefined in Module B.

### Actual Behavior

In our Electron application, we have a module (Module A) which uses child_process.fork() to fork another module (Module B). Module B is supposed to send a message back to Module A using process.send(). However, since updating to Electron v28.1.2, process.send is undefined in Module B, preventing it from sending messages back to Module A.

### Additional Information

- This issue appeared after updating to Electron v28.1.2.
- The same code works as expected in earlier versions of Electron.

**Comment by codebytere** [comment]
https://github.com/electron/electron/compare/v28.1.1...v28.1.2 - potentially https://github.com/electron/electron/commit/7989f02a433b432b1498d562d90e294ae7bdc05a 

I'm not sure if this was an intended side effect of the above commit, but it may have been for security purposes. @zcbenz would be able to answer that!

**Comment by ravicious** [comment]
https://github.com/electron/electron/blob/d567a540c4d0074365ac9df55a1d76816712219e/shell/common/mac/codesign_util.h#L13-L20

https://github.com/electron/electron/blob/d567a540c4d0074365ac9df55a1d76816712219e/shell/app/node_main.cc#L115-L132

In our case, the main process of a signed macOS app forks a .js file from app.asar with some extra arguments. Are those processes considered to have different process signatures because the main process is `Contents/MacOS/<app name>` and the forked one is actually `Contents/Frameworks/<app name> Helper.app/Contents/MacOS/<app name> Helper`?

**Comment by theDevelopper** [comment]
If this change is on purpose, then it seems like a breaking change to me.

I am in the same boat, as now the communication with any child process in my app is not working anymore. I tried to use UtilityProcess in the past but could not make it work. Does this change affect UtilityProcess too?

**Comment by jwetzell** [comment]
Experiencing this as well.

**Comment by koush** [comment]
This is affected my app as well. child_process.fork.send is no longer received by the child process.

**Comment by zcbenz** [comment]
I'm certain this is caused by https://github.com/electron/electron/pull/40770 dropping envs used for IPC communication. I'll create a quick fix first, but it would help me understand the root issue if you can answer following questions:

1. Where do you run `fork` (main/renderer/unity process)?
2. Is the app signed (or ad-hoc signed)?
3. If the app is signed, does the behavior happen if you run the unsigned app?

**Comment by theDevelopper** [comment]
> 1. Where do you run `fork` (main/renderer/unity process)?
> 2. Is the app signed (or ad-hoc signed)?
> 3. If the app is signed, does the behavior happen if you run the unsigned app?

For me, it happens on macOS:
1. fork done in main process
2. app is signed and notarized
3. it does not happen with unsigned version

**Comment by ravicious** [comment]
Our app works the same way.

1. Fork in the main process.
2. The app is signed and notarized.
3. IPC works fine in the unsigned version of the app.

**Comment by jwetzell** [comment]
Yep same thing here main process, signed does not work and unsigned worked.

**Comment by seydx** [comment]
Same

**Comment by jkleinsc** [comment]
Fixed via #41079.
